### PR TITLE
ENH: Adds base work on a view controller - depends on #407

### PR DESCRIPTION
**Pull request by ElDeveloper**
Note that the approach that I took here is rather different than what we originally planned. Essentially what will result of this is class `AttributeViewController` that can take care of 'color', 'opacity' or 'visibility', since these are soooooo similar between each other, I took this approach, however I'm down to hear whatever you guys think would be better.

The main relevant file is `view-controller.js`.

~~Note, this PR depends on #407~~

### CODE REVIEW

**Comment by emperor-helper** [comment]
The test build for this pull request can be found here: http://emperor.microbio.me/pull_413/make_emperor/

**Comment by ElDeveloper** [comment]
This PR is open for bike-shedding! Note that the generic view controller hasn't yet been integrated with the rest of the system but, this is what I had in mind, instead of a hierarchy of rather redundant classes.

**Comment by ElDeveloper** [comment]
I just updated this PR to integrate the generic view controller in the main application (thus the huge blob of deletions), please have a look and let me know if this makes sense. If we keep following this pattern, all remaining tabs in the app should become either unique classes or part of the `AttributeController` class. I would advise looking at the interface for the `AttributeController`, mainly the way the class is being initialized, it feels too tightly coupled with the rest of the objects, and there are a few methods missing, as usual run `./build.py XXX` and open new-emperor.html to test this out.

**Comment by emperor-helper** [comment]
Anyone wants to take a pass, or should I start? :ghost:

**Comment by antgonza** [comment]
:+1: are we doing single or multiple reviews?

**Comment by ElDeveloper** [comment]
We are doing as many reviews as possible, but at least 1 :). In this
case @mortonjt, @josenavas and me discussed more in-depth about the
design, and we decided to change this, though I haven't actually done
that. There's a bit more of discussion that needs to happen before that.

I'll be better at updating PRs with offline discussions!

File: emperor/support_files/js/view-controllers.js [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+                                minWidth: scope.SLICK_WIDTH,
+                                editor: ColorEditor,
+                                formatter: ColorFormatter};
+    }
+    else if (scope.attrName === 'opacity') {
+      // TODO
+    }
+    else if (scope.attrName === 'visibility') {
+      // TODO
+    }
```
Comment (mortonjt) [inline emperor/support_files/js/view-controllers.js]: This will have to change, since we opted for multiple subclasses instead of one giant superclass

**Comment by ElDeveloper** [comment]
Closing this PR, in favor of a new one that @mortonjt and me will work on.

:church: PR #143.

**Pull request closed without merging**